libtracepoint-control-cpp/include/tracepoint/TracepointSession.h

```diff
@@ -247,6 +248,91 @@ namespace tracepoint_control
         uint32_t m_sampleType;
     };
 
+    /*
+    Configuration settings for TracepointSession::SavePerfDataFile.
+
+    Example:
+
+        error = session.SavePerfDataFile(
+            "perf.data",
+            TracepointSavePerfDataFileOptions().OpenMode(S_IRUSR | S_IWUSR));
+    */
+    class TracepointSavePerfDataFileOptions
```

Comment:
Note: Instead of passing a bunch of parameters to the SavePerfDataFile method, pass an optional Options parameter with various options.

- OpenMode (permissions to apply to the created perf.data file)
- TimestampFilter (only write events within a time range)
- TimestampRange (receives the timestamps of the first and last events written)